packages/react-google-maps-api/src/useLoadScript.ts

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react'
+
+import { injectScript } from './utils/injectscript'
+import { preventGoogleFonts } from './utils/prevent-google-fonts'
+
+import { isBrowser } from './utils/isbrowser'
+import { LoadScriptProps, defaultLoadScriptProps } from './LoadScript'
+
+export const useLoadScript = ({
+  id = defaultLoadScriptProps.id,
+  version = defaultLoadScriptProps.version,
+  googleMapsApiKey,
+  language,
+  region,
+  libraries,
+  preventGoogleFontsLoading,
+  onError,
+  onLoad,
+  onUnmount
+}: LoadScriptProps) => {
+  const isMounted = React.useRef(false)
+  const [isLoaded, setLoaded] = React.useState(false)
+
+  React.useEffect(() => {
+    isMounted.current = true
+    return () => {
+      isMounted.current = false
+      if (onUnmount) {
+        onUnmount()
+      }
+    }
+  }, [])
+
+  React.useEffect(() => {
+    if (isBrowser && preventGoogleFontsLoading) {
+      preventGoogleFonts()
+    }
+  }, [preventGoogleFontsLoading])
+
+  const url = makeUrl()
+
+  React.useEffect(() => {
+    if (!isBrowser) {
+      return
+    }
+
+    const setLoadedIfMounted = (loaded: boolean) => {
+      if (isMounted.current) {
+        setLoaded(loaded)
+      }
+    }
+
+    // @ts-ignore
+    if (window.google) {
+      console.error('google api is already presented')
+      setLoadedIfMounted(true)
+      return
+    }
+
+    injectScript({ id, url })
+      .then(() => {
+        setLoadedIfMounted(true)
+        if (onLoad) {
+          onLoad()
+        }
+      })
+      .catch(err => {
+        if (onError) {
+          onError(err)
+        }
+        console.warn(`
+        There has been an Error with loading Google Maps API script, please check that you provided correct google API key to <LoadScript /> (${googleMapsApiKey})
+        Otherwise it is a Network issues.
+      `)
+        console.error(err)
+      })
+
+    return () => {
+      setLoadedIfMounted(false)
+    }
+  }, [id, url])
```

Comment:
I should probably include the `onLoad` and `onError` here as well, but it's a kinda bad idea to be re-injecting script if a parent is not having `useCallback` for these. So I think I will have to keep those in a ref otherwise it would see only first instances. Any idea of how to tackle it better?

Reply:
Nevermind, figured out a better way with separate useEffect calls. Although it might be better to keep those callbacks out of the hook and use them only for a component variant.